app/src/main/java/de/danoeh/antennapod/fragment/FeedInfoFragment.java

```diff
@@ -218,18 +235,52 @@ private void showFeed() {
             lblAuthor.setVisibility(View.GONE);
             txtvAuthor.setVisibility(View.GONE);
         }
-        if (!TextUtils.isEmpty(feed.getLanguage())) {
-            txtvLanguage.setText(LangUtils.getLanguageString(feed.getLanguage()));
+        if (podcastStatistics != null) {
+            txtvStatistics.setText(getString(R.string.podcast_listendes_for)
+                    + " "
+                    + Converter.shortLocalizedDuration(getContext(), podcastStatistics.timePlayed)
+                    + "\n"
+                    + getString(R.string.podcast_on_device)
+                    + " "
+                    + Formatter.formatShortFileSize(getContext(), podcastStatistics.totalDownloadSize)
+                    + " • "
+                    + String.format(Locale.getDefault(), "%d%s",
+                    podcastStatistics.episodesDownloadCount, getString(R.string.episodes_suffix)));
         } else {
-            lblLanguage.setVisibility(View.GONE);
-            txtvLanguage.setVisibility(View.GONE);
+            lblStatistics.setVisibility(View.GONE);
+            txtvStatistics.setVisibility(View.GONE);
         }
+
         txtvUrl.setText(feed.getDownload_url() + " {fa-paperclip}");
         Iconify.addIcons(txtvUrl);
-
         getActivity().invalidateOptionsMenu();
     }
 
+    private void loadStatistics() {
+        if (disposable != null) {
+            disposable.dispose();
+        }
+
+        disposable =
+                Observable.fromCallable(() -> {
+                    List<StatisticsItem> statisticsData = DBReader.getStatistics();
+                    Collections.sort(statisticsData, (item1, item2) ->
+                            CompareCompat.compareLong(item1.totalDownloadSize, item2.totalDownloadSize));
+                    return statisticsData;
+                })
+                        .subscribeOn(Schedulers.io())
+                        .observeOn(AndroidSchedulers.mainThread())
+                        .subscribe(result -> {
+                            for (StatisticsItem item : result) {
+                                if (item.feed.getId() == feed.getId()) {
+                                    podcastStatistics = item;
+                                    break;
+                                }
+                            }
```

Comment:
I would move this loop up to the `fromCallable` section. That way, searching is done in a background thread instead of the main UI thread.